frontend/class-opengraph.php

```diff
@@ -505,11 +512,15 @@ public function description( $echo = true ) {
 			}
 		}
 
-		if ( is_singular() ) {
-			$ogdesc = WPSEO_Meta::get_value( 'opengraph-description' );
+		$is_posts_page = is_home() && ! is_front_page();
+
+		if ( is_singular() || $is_posts_page ) {
+			$post_id = ( $is_posts_page ) ? get_option( 'page_for_posts' ) : get_the_ID();
+			$post    = get_post( $post_id );
```

Comment:
I will ponder de-duplication... It was a little hard to settle on something since Facebook and Twitter do essentially same thing, but are written (and need to be tested) differently.